lib/paperclip/has_attached_file_extensions.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Paperclip
+  module HasAttachedFileExtensions
+    # Rails 6 doesn't support (or silently ignore) :on
+    def add_active_record_callbacks
+      name = @name
+      @klass.send(:after_save) { send(name).send(:save) }
+      @klass.send(:before_destroy) { send(name).send(:queue_all_for_delete) }
+      @klass.send(:after_destroy) { send(name).send(:flush_deletes) }
+    end
+  end
+end
+
+Paperclip::HasAttachedFile.prepend(Paperclip::HasAttachedFileExtensions)
```

Comment:
I'm not really comfortable with that change, why is it needed?

Reply:
It was a hack to support `:on` with `after_commit` after the rails 6 upgrade, but it turns out the callback isn't needed anyway, so I've removed it along with the workaround